zerver/lib/rate_limiter.py

```diff
@@ -247,6 +244,9 @@ def is_ratelimited(cls, entity_key: str, rules: List[Tuple[int, int]]) -> Tuple[
                 blocking_ttl = int(blocking_ttl_b)
             return True, blocking_ttl
 
+        if len(rules) == 0:
+            return False, 0.0
+
```

Comment:
Would it be better to move the manual blocking rule up, rather than moving this check down?  We'd avoid an unnecessary connection to redis when we only we're not going to do there.

Reply:
Possibly, but `rate_limiter: Handle edge case where rules list may be empty.` deletes this anyway, so probably not worth spending time on this detail for now, unless we decide not to merge that commit.